src/mapreduce.jl

```diff
@@ -29,7 +16,7 @@ end
     newT = :(Core.Inference.return_type(f, Tuple{$(eltypes...)}))
     return quote
         @_inline_meta
-        @inbounds return similar_type(typeof(_first(a...)), $newT)(tuple($(exprs...)))
+        @inbounds return similar_type(typeof(_first_static(a...)), $newT, Size(S))(tuple($(exprs...)))
```

Comment:
You're right, I wasn't sure about this change.  I'm happy to change it back to the first argument.